include/pipert/UDPConnection.h

```diff
@@ -0,0 +1,47 @@
+#ifndef _UDPCONNECTION_H_
+#define _UDPCONNECTION_H_
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+
+namespace pipert {
+
+/// This class is responsible for UDP communication.
+class UDPConnection {
+ public:
+  /// Constructs new UDP connection instance.
+  ///
+  /// \param remote_port UDP connection port
+  /// \param remote_address UDP connection ip address
+  UDPConnection(int remote_port, const char* remote_address);
+
+  /// Constructs new UDP connection instance.
+  ///
+  /// \param binding_port UDP connection port
+  UDPConnection(int binding_port);
+
+  /// Destructor of UDPConnection.
+  ~UDPConnection();
+
+  /// Returns whether the UDP connection was properly initialized.
+  bool IsConnected() const { return socket_filedesc_ != -1; }
+
+  /// Sends the data through UDP connection.
+  /// \param buffer holds the buffer that needs to be sent.
+  /// \param size holds the buffer size.
+  void Send(void* buffer, int size);
+
+  /// Receives data from UDP connectom.
+  /// \param buffer provides the buffer in which to receive data.
+  /// \param size holds the buffer size.
+  /// \return The number of bytes received, or -1 if an error occurred.
+  int Receive(void* buffer, int size);
+
+ private:
+  int socket_filedesc_;
```

Comment:
I think, we will have a problem with this class. It has a default copy constructor and you are copying it at places. This means, that this open file descriptor will be used from multiple places. Is it ok this way? Can it happen that multiple threads call an operation on the same file descriptor simultaneously? What should be the expected behavior? And what happens when one object destructs? The file descriptor will be closed also under the other objects. Can you check this?

Reply:
This question is still valid I guess.

Reply:
In a later commit, I deleted the copy constructor to avoid copying the file descriptor. Check [9894fe3](https://github.com/gerazo/pipert/pull/46/commits/9894fe32f4e2b3e28f70fd3d2f1744a1bb4c6cf4) commit.

Reply:
Sorry, my bad. default and delete looks the same :)